.github/workflows/main.yml

```diff
@@ -5,255 +5,122 @@ on:
     - cron: "40 2 * * *"
   push:
     branches:
-      - '**'
+      - 'main'
     tags:
       - version-*
   pull_request:
+  workflow_dispatch:
 
 jobs:
-        
-  build-cpp:
-    runs-on: ubuntu-18.04
+  build-cpp-gcc_clang:
+    strategy:
+      matrix:
+        compiler: ["gcc", "clang"]
+        config: ["Debug", "Release"]
+        version: ["latest", "min"]
+        include:
+          - version: "latest"
+            os: "ubuntu-latest"
+          - version: "min"
+            os: "ubuntu-18.04"
+    runs-on: ${{ matrix.os }}
+    continue-on-error: ${{ matrix.compiler == 'clang' && matrix.version == 'latest' }} # our jsoncpp version does not build on clang above version 9
```

Comment:
I dont know if you are still working on clang since a job was failing but I suppose this won't be fixed here (which is OK ;-)) But if thats really the case, please make it an issue.

Reply:
#68